am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GrantAccessIntegrationTest.java

```diff
@@ -21,9 +30,43 @@ public void setupTest() {
     }
 
     @Test
-    public void createQuery_whenCreatingResourceAccess_ResourceAccessAppearsInDatabase() {
-        ams.createResourceAccess(createRecord(resourceId, ACCESSOR_ID, EXPLICIT_READ_CREATE_UPDATE_PERMISSIONS));
+    public void grantAccess_emptyPermissionsMap_shouldThrowException() {
+        assertThatExceptionOfType(IllegalArgumentException.class).isThrownBy(() ->
+            ams.grantExplicitResourceAccess(createRecord(resourceId, ACCESSOR_ID, EMPTY_ATTRIBUTE_PERMISSIONS)))
+            .withMessage("Attribute permissions cannot be empty");
+    }
+
+    @Test
+    public void grantAccess_whenCreatingResourceAccess_ResourceAccessAppearsInDatabase() {
+        ams.grantExplicitResourceAccess(createRecord(resourceId, ACCESSOR_ID, SINGLE_ATTRIBUTE_PERMISSION));
+
+        assertThat(countResourcesById(resourceId)).isEqualTo(1);
+    }
+
+    @Test
+    public void grantAccess_whenCreatingResourceAccess_EmptyAttribute() {
```

Comment:
By the way we stop prefixing test method names with service method name since we separated tests into independent test classes recently. Removing `grantAccess_` for example will leave us more space for better test names.